### PR TITLE
Refactor printing page

### LINKED ISSUE #253: Printing iframe reload status
Embed updating system crashed plugin on my computer a few times. Maybe it's my laptop that's slow but maybe add something to indicate the embed is loading and disable everything during that time. 

Never done this before might need help

### LINKED ISSUE #433: 2D Printing page code refactor
There is too much jumbled up unnecessary code in the 2D Printing page. Split the file into multiple files so that others can read it easier

### PR DESCRIPTION
**Pull request by jerrylee17**
Resolves #433 
Resolves #253 

Printing page code refactor
Print preview now has a loading button when the page is refreshed

### CODE REVIEW

**Comment by jerrylee17** [comment]
Just keeping my links here: 
https://medium.com/@josh.j.pearson/handling-iframe-loading-in-react-57f044a9d0fa
https://stackoverflow.com/questions/8626638/how-to-display-loading-message-when-an-iframe-is-loading

File: src/Pages/2DPrinting/2D-printing.css [hunk 1]
```diff
@@ -75,3 +75,16 @@
...
+  top: 50%;
+  right: 50%;
+}
+
+.spinner-border {
+  color: blue;
+  width: 5rem;
+  height: 5rem;
+  border-width: 0.45em;
+}
```
Comment (hsorathia) [inline src/Pages/2DPrinting/2D-printing.css:90]: new line at eof

File: src/Pages/2DPrinting/2DPrinting.js [hunk 2]
```diff
@@ -49,15 +54,87 @@ export default function Printing(props) {
...
+    color: 'success',
+    onClick: () => {
+      setConfirmModal(!confirmModal);
+    },
+    disabled: !canPrint,
+    text: 'Print!'
+  };
+
+  const previewModalProps = {
+    isOpen: previewModal,
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DPrinting.js:101]: i dont think this is a valid prop, see chrome output
image

File: src/Pages/2DPrinting/2DPrinting.js [hunk 3]
```diff
@@ -49,15 +54,87 @@ export default function Printing(props) {
...
+    },
+    text: 'Continue'
+  };
+
+  const backButtonProps = {
+    color: 'danger',
+    onClick: () => {
+      setPreviewModal(false);
+      setPages(false);
+      let arr = new Set(range(1, numPages + 1));
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DPrinting.js:85]: maybe add a comment to explain what this does

File: src/Pages/2DPrinting/2DPrinting.js [hunk 4]
```diff
@@ -160,191 +236,133 @@ export default function Printing(props) {
...
-            }}
-            onClick={() => {
-              setConfirmModal(!confirmModal);
-            }}
-            disabled={!canPrint}
-          >
-            Print!
-          </Button>
+          <Button {...backButtonProps}> Back </Button>
+          <Button {...printButtonProps}> Print! </Button>
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DPrinting.js:365]: Is there a way we can tell the user the print was successful or not? as of now it goes quiet after we submit a request.

File: src/Pages/2DPrinting/2DPrinting.js [hunk 5]
```diff
@@ -160,36 +274,12 @@ export default function Printing(props) {
...
-      <Modal
-        isOpen={previewModal}
-        toggle={() => {
-          setPreviewModal(!previewModal);
-        }}
-        size='xl'
-      >
+      <Button {...continueButtonProps}> Continue </Button>
+
+      <Modal {...previewModalProps}>
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DPrinting.js:282]: make this modal its own component in the `2DPrinting` directory

File: src/Pages/2DPrinting/2DPrinting.js [hunk 6]
```diff
@@ -49,15 +55,90 @@ export default function Printing(props) {
     props.user.pagesPrinted
   );
   const [previewDisplay, setPreviewDisplay] = useState('');
+  const [loadPreview, setLoadPreview] = useState(true);
+  const [statusModal, setStatusModal] = useState(false);
+  const [printStatus, setPrintStatus] = useState('');
+
   const previewLabels = {
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DPrinting.js:62]: you might as well throw all these `const` definitions into `2DComponents.js`

Comment (jerrylee17) [reply src/Pages/2DPrinting/2DPrinting.js:62]: But they're used in the loading mechanism/printing mechanism of the actual page and are modified by other methods in the same function...

Comment (evanugarte) [reply src/Pages/2DPrinting/2DPrinting.js:62]: ok nvm

**Comment by evanugarte** [comment]
@jerrylee17 can remove the body to just be blank
instead of:
image

**Comment by evanugarte** [comment]
@jerrylee17 also for the confirmation modal just make it have one button `Ok`, there isnt a point in having 2
image

**Comment by evanugarte** [comment]
@jerrylee17 say sum like
```
Sorry! Our printing system is down at the moment, please see an officer or try again later.
```
image

**Comment by jerrylee17** [comment]
Going to put a custom commit message when it merged into master so no need to squash

File: src/Pages/2DPrinting/2DComponents.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+export const PrintIcon = `Drag & Drop or Touch Here <br />
+<svg aria-hidden='true' viewBox='0 0 512 512' width='40%'>
+<path
+  d='M399.95 160h-287.9C76.824 160 48 188.803 48
+  224v138.667h79.899V448H384.1v-85.333H464V224c0-35.197-28.825-64-64.05-64zM352
+  416H160V288h192v128zm32.101-352H127.899v80H384.1V64z'
+  />
+</svg>`;
+
+export const PrintInfo = `Welcome to printing! Click the icon below
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DComponents.js:37]: change the ticks to single quotes

File: src/Pages/2DPrinting/2DComponents.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      <ModalFooter>
+        <Button color={buttonColor} onClick={handleConfirmation}>
+          {confirmText}
+        </Button>
+      </ModalFooter>
+    </Modal>
+  );
+}
+
+export const PrintIcon = `Drag & Drop or Touch Here <br />
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DComponents.js:28]: change the ticks to single quotes

File: src/Pages/2DPrinting/2DComponents.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  d='M399.95 160h-287.9C76.824 160 48 188.803 48
+  224v138.667h79.899V448H384.1v-85.333H464V224c0-35.197-28.825-64-64.05-64zM352
+  416H160V288h192v128zm32.101-352H127.899v80H384.1V64z'
+  />
+</svg>`;
+
+export const PrintInfo = `Welcome to printing! Click the icon below 
+and upload your file. Each member can print up to 30 pages a week.`;
+
+export const failPrintStatus = `Sorry! Our printing system is down
```
Comment (evanugarte) [inline src/Pages/2DPrinting/2DComponents.js:40]: change the ticks to single quotes

**Pull request merged**